### DISCUSSION TITLE
WHIPs 12 and 13: Initial governance and voting

### DISCUSSION THREAD

**Discussion by urosvk** [wihi-labs/WHIP#21] [Ideas]
- Authors: [Mark Ballandies](https://twitter.com/BallandiesMC)
- Start Date: 2024-02-15
- Category: Community, Data
----

## Summary

WiHi's envisioned voting mechanism is outlined in WHIP-0007, WiHi's decentralized organization is described in WHIP-0002, and WiHi's instantiated legal structure for this organization is illustrated in WHIP-0008. This WHIP illustrates the initial mechanism with which WiHi will reach decisions on WHIPs and voting items within the legal organization (e.g. such as electing the governance committee). WHIP-0013 implements the voting mechanism of the initial governance illustrated in WHIP-0012.

Links to WHIPs: 

https://github.com/wihi-labs/WHIP/blob/main/0012-initial-governance.md
https://github.com/wihi-labs/WHIP/blob/main/0013-initial-voting.md

**Comment by urosvk** [comment]
Fine with 12 as-is

For 13:
- "New units of vote power are assigned at the end of each month and decided on by the voting mechanism illustrated in this WHIP according to the previous vote power distribution." How is this to be done? Didn't see it in the implementation section. This is important because it needs to take into account that voting power is represented by SPL tokens, e.g., there's only so much you can do with token authority.
- Does it need to be a test token? It seems like voting is to be distributed according to some mechanism and it is the mechanism that changes, not the token.
- Don't like the name "voting power"; would prefer to prepend v- to whatever our token name will be, i.e., vWIHI

**Comment by ballandiesmark** [comment]
1. The following two bullet points are added to the implementation: - the votes are conducted via the Realms proposal mechanism. - Also the creation and assignment of new vWiPo is performed via this proposal mechanism. 
2. I called it a test token, because i) our WiHi token is currently envisioned to give eventually access to the governance in the system, ii) it does not explicitly implement the envisioned square root voting, but "only" implicitly (because of the limitations of Realms), iii) it is dependent on the off-chain cap table. Nevertheless, I reformulated the sentence to "Vote power is a temporary token to facilitate the initial governance (WHIP-12)".
4. What do you think of "vWiPo"?

**Comment by chiumichael** [comment]
- vWiPo and WiPo need to be on the testnet. This is consistent with their status as test tokens and saves the DAO money as Solana prices continue to increase. An additional benefit is that any mistakes that happen can stay on the testnet.
- How do the tokens in the initial voting jive with the tokenomics that the core team has recently been discussing? 
- `Motivation` section should (eventually?) be filled in with prior work / prior lit / comparison to what other projects do in terms of voting, etc. I think there was a non-trivial amount of work in writing this WHIP and this is something that the community (speaking from  can learn from.

**Reply by ballandiesmark** [reply]
point 1: Disagree, we should test the infrastructure as realistically as possible. This goes with the best practice in the industry to deploy points to the main networks. In particular, supporters should see the WiPOs next to their other token holdings. Also, user experience won’t be great, if they have to go to (for instance under phantom) to the developer settings and enable the test network. This could be a significant barrier for non-web3 people to experience/ test WiHis platform. Considering the low cost of Solana transactions, I suggest to deploy to mainnet directly.

point 2: In the initial voting we have one token, the vWiPo. It is setup such its mechanism are aligned with the voting sketched out in WHIP-0007/WHIP-0008, the association articles and our founders pool mechanism. Hence, it is the best way to organize voting atm as it represents our current status-quo with the small extension that we decentralize voting to all founders. In case tokenomics changes, this will be subject to the illustrated voting mechanism. In particular, we can vote on changing the nature/ characteristics/ type of the vote token. I do not see a problem here (pls correct me!). 

point 3: according to WHIP-0001, the Motivation section contains the motivation of the authors for proposing the improvement, e.g. motivation could be to have a process in place to finalize the open WHIPs. This will be filled out asap (The current version of the WHIP was created in the hope to get quick feedback). What you are asking for could be integrated into the rationale section that illustrates the rational of design decisions within the specification. I will extend that section by providing further rationale to the chosen specification (also with regards to the previous point), which might contain references to literature. Nevertheless, Im not planning to do a literature review here because the suggested mechanism is the result from the current boundary conditions, e.g. WHIP-0007/8, thus highlighting those and showing how WHIP-12/13 are in accordance should be sufficient.
